cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionUtils.java

```diff
@@ -132,13 +132,22 @@ public static ConnectorContext getConnectorContext(ServicePluginConfigurer plugi
    */
   public static ConnectorSpec filterSpecWithPluginNameAndType(ConnectorSpec spec, @Nullable String pluginName,
                                                               @Nullable String pluginType) {
-    if (Strings.isNullOrEmpty(pluginName) || Strings.isNullOrEmpty(pluginType)) {
+    if (Strings.isNullOrEmpty(pluginName) && Strings.isNullOrEmpty(pluginType)) {
       return spec;
     }
     ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
     specBuilder.setSchema(spec.getSchema());
     for (PluginSpec pluginSpec : spec.getRelatedPlugins()) {
-      if (pluginType.equals(pluginSpec.getType()) && pluginName.equals(pluginSpec.getName())) {
+      if (Strings.isNullOrEmpty(pluginName)) {
+        if (pluginType.equalsIgnoreCase(pluginSpec.getType())) {
+          specBuilder.addRelatedPlugin(pluginSpec);
+        }
+      } else if (Strings.isNullOrEmpty(pluginType)) {
```

Comment:
I don't think you need `else` here, and you can combine the nested if condition into 
```
if (Strings.isNullOrEmpty(pluginName) && pluginType.equalsIgnoreCase(pluginSpec.getType()) {
  specBuilder.addRelatedPlugin(pluginSpec);
  continue;
}
```